shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -16,23 +16,55 @@
 
 package com.nvidia.spark.rapids.shims.spark310
 
+import java.nio.{ByteBuffer, ByteOrder}
+
 import scala.collection.JavaConverters._
 import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
 
 import ai.rapids.cudf._
 import com.nvidia.spark.rapids._
 import com.nvidia.spark.rapids.GpuColumnVector.GpuColumnarBatchBuilder
 import com.nvidia.spark.rapids.RapidsPluginImplicits._
+import org.apache.avro.util.ByteBufferInputStream
+import org.apache.hadoop.conf.Configuration
+import org.apache.parquet.HadoopReadOptions
+import org.apache.parquet.hadoop.{ParquetFileReader, ParquetInputFormat}
+import org.apache.parquet.io.{DelegatingSeekableInputStream, InputFile, SeekableInputStream}
 
 import org.apache.spark.TaskContext
 import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression}
+import org.apache.spark.sql.catalyst.expressions.codegen.GenerateUnsafeProjection
 import org.apache.spark.sql.columnar.{CachedBatch, CachedBatchSerializer}
+import org.apache.spark.sql.execution.datasources.parquet.{ParquetReadSupport, ParquetToSparkSchemaConverter, ParquetWriteSupport}
+import org.apache.spark.sql.execution.datasources.parquet.rapids.ParquetRecordMaterializer
 import org.apache.spark.sql.internal.SQLConf
-import org.apache.spark.sql.types.{StructField, StructType}
+import org.apache.spark.sql.internal.SQLConf.LegacyBehaviorPolicy
+import org.apache.spark.sql.types._
 import org.apache.spark.sql.vectorized.ColumnarBatch
 import org.apache.spark.storage.StorageLevel
+import org.apache.spark.util.SerializableConfiguration
+
+class ByteArrayInputFile(buff: Array[Byte]) extends InputFile {
+
+  override def getLength: Long = buff.length
+
+  override def newStream(): SeekableInputStream = {
+    // ParquetFileReader reads in little endian! :shrugs
+    val byteBuffer = ByteBuffer.wrap(buff).order(ByteOrder.LITTLE_ENDIAN)
+    new DelegatingSeekableInputStream(new ByteBufferInputStream(List(byteBuffer).asJava)) {
+      override def getPos: Long = byteBuffer.position()
+
+      override def seek(newPos: Long): Unit = {
+        assert(newPos <= Int.MaxValue && newPos >= Int.MinValue)
```

Comment:
I don't think this should be an assert.  It will be quite difficult to debug if the assert doesn't fire in a regular run in the wild and this silently truncates the desired seek position.